solr/server/build.gradle

```diff
@@ -30,39 +30,62 @@ configurations {
 }
 
 dependencies {
-  api('org.eclipse.jetty:jetty-alpn-java-server', {
-    exclude group: "org.eclipse.jetty.alpn", module: "alpn-api"
-  })
-
+  permitUnusedDeclared 'io.dropwizard.metrics:metrics-core'
   api('io.dropwizard.metrics:metrics-core', {
     exclude group: "com.rabbitmq", module: "amqp-client"
   })
+  permitUnusedDeclared 'io.dropwizard.metrics:metrics-graphite'
   api('io.dropwizard.metrics:metrics-graphite', {
     exclude group: "com.rabbitmq", module: "amqp-client"
   })
+  permitUnusedDeclared 'io.dropwizard.metrics:metrics-jetty9'
   api 'io.dropwizard.metrics:metrics-jetty9'
+  permitUnusedDeclared 'io.dropwizard.metrics:metrics-jvm'
   api 'io.dropwizard.metrics:metrics-jvm'
+  permitUnusedDeclared 'io.dropwizard.metrics:metrics-jmx'
   api 'io.dropwizard.metrics:metrics-jmx'
```

Comment:
Maybe these shouldn't be `api` in the first place?

Reply:
So to be completely honest with you - I don't quite understand `solr/server/build.gradle` and how it is different than like core with all the Jetty dependencies. I don't even know if the `api` here even makes any difference. @janhoy on SOLR-16007 mentioned `server/lib` and `WEB-INF/lib` and I don't quite understand the difference between the two right now. So I don't really feel super comfortable with changing from `api` to anything else until I can understand why this was configured this way. (there is some weird webapp -> core -> server dependency stuff going on)